docs/Middleware-Guide.md

```diff
@@ -0,0 +1,82 @@
+
+The following figure illustrates the above flow: 
+![Staker opting-in](./images/staker_opting.png)
+
+
+### *Staker withdraws*
+In order for a staker, who has opted-in to a middleware via delegation, wants to partially withdraw some of its stake or completely stop participating, it has to undelegate its stake from the current operator. Under current design of delegation, whenever the staker partially withdraw some of its stake or completely stop participating, then it affects all the middlewares uniformly that its operator is participating in. The sequential flow is as follows:
+1. The staker first has to queue its withdrawal request with EigenLayer. The staker can place this request by calling  `queueWithdrawal(..)` in the EigenLayer's `StrategyManager.sol`. On successful call to this, the event `WithdrawalQueued(..)` is emitted.
+2. The staker then has to notify all the middlewares that its operator has opted in about its intention to partially withdraw some of its stake or completely stop participating and also record the latest time in the EigenLayer contracts until which the staker is obliged to participate with the current stake. To do this, the staker calls on the `updateStakes(..)` function in the Registry contract that calls `recordStakeUpdate(..)` in the middleware's ServiceManager contract which in turn accesses `recordStakeUpdate(..)` in the EigenLayer's Slasher contract.  On successful execution of this call, the event `MiddlewareTimesAdded(..)` is emitted.
+3. After the latest time until which the staker is obliged to participate with the current stake passes, the staker now can complete its queued withdrawal by calling on the `completeQueuedWithdrawal(..)` function in `StrategyManager.sol`.
+
+The following figure illustrates the above flow: 
+![Staker deregistering](./images/staker_withdrawing.png)
+
+
+### *Slashing*
+As mentioned above, EigenLayer is built to support slashing as a result of an on-chain-checkable, objectively attributable action. In order for a middleware to be able to slash an operator in an objective manner, the middleware needs to deploy a DisputeResolution contract where anyone would be able to raise a challenge against an EigenLayer operator for its adversarial action. On successful challenge, the DisputeResolution contract calls the `freezeOperator(..)` function in ServiceManager contract which in turn accesses the `freezeOperator(..)` function in the EigenLayer's Slasher contract. On successful execution of this API, the event `OperatorFrozen(..)` is emitted.  
```

Comment:
"where anyone would be able to" => "which anyone can call to"

Reply:
check out how my suggested change works:
now:
calls the `freezeOperator(..)` function in ServiceManager contract which in turn accesses the `freezeOperator(..)` function in the EigenLayer's Slasher contract.
with change:
calls the `ServiceManager.freezeOperator(..)`; the ServiceManager in turn calls `Slasher.freezeOperator(..)` to freeze the operator in EigenLayer

Reply:
change
"On successful execution of this API, the event `OperatorFrozen(..)` is emitted."
to something like
"EigenLayer's Slasher contract emits a `OperatorFrozen(..)` event whenever an operator is (successfully) frozen"

Reply:
done: [4b68a0a](https://github.com/Layr-Labs/eigenlayer-contracts/pull/16/commits/4b68a0a046737e7089d054a8e8451f1f71c12848)